main.go

```diff
@@ -0,0 +1,176 @@
+func (v *VersionCmd) Run(*CLI) error   { return delegateToCobraSystem("version", v.Args) }
+func (a *AgentCmd) Run(*CLI) error     { return delegateToCobraSystem("agent", a.Args) }
+func (a *ArtifactsCmd) Run(*CLI) error { return delegateToCobraSystem("artifacts", a.Args) }
+func (b *BuildCmd) Run(*CLI) error     { return delegateToCobraSystem("build", b.Args) }
+func (c *ClusterCmd) Run(*CLI) error   { return delegateToCobraSystem("cluster", c.Args) }
+func (j *JobCmd) Run(*CLI) error       { return delegateToCobraSystem("job", j.Args) }
+func (p *PackageCmd) Run(*CLI) error   { return delegateToCobraSystem("package", p.Args) }
+func (p *PipelineCmd) Run(*CLI) error  { return delegateToCobraSystem("pipeline", p.Args) }
+func (u *UserCmd) Run(*CLI) error      { return delegateToCobraSystem("user", u.Args) }
+func (a *ApiCmd) Run(*CLI) error       { return delegateToCobraSystem("api", a.Args) }
+func (c *ConfigureCmd) Run(*CLI) error { return delegateToCobraSystem("configure", c.Args) }
+func (i *InitCmd) Run(*CLI) error      { return delegateToCobraSystem("init", i.Args) }
+func (u *UseCmd) Run(*CLI) error       { return delegateToCobraSystem("use", u.Args) }
+
+func delegateToCobraSystem(command string, args []string) error {
+	originalArgs := os.Args
+	defer func() { os.Args = originalArgs }()
+
+	os.Args = append([]string{os.Args[0], command}, args...)
+
+	if code := runCobraSystem(); code != 0 {
+		os.Exit(code)
+	}
+	return nil
+}
+
+func runCobraSystem() int {
+	f, err := factory.New(version.Version)
+	if err != nil {
+		handleError(bkErrors.NewInternalError(err, "failed to initialize CLI", "This is likely a bug", "Report to Buildkite"))
+		return bkErrors.ExitCodeInternalError
+	}
+
+	rootCmd, err := root.NewCmdRoot(f)
+	if err != nil {
+		handleError(bkErrors.NewInternalError(err, "failed to create commands", "This is likely a bug", "Report to Buildkite"))
+		return bkErrors.ExitCodeInternalError
+	}
+
+	rootCmd.SetContext(context.Background())
+	rootCmd.SilenceErrors = true
+
+	if err := rootCmd.Execute(); err != nil {
+		handleError(err)
+		return 1
+	}
+	return 0
+}
+
+func handleError(err error) {
+	bkErrors.NewHandler().Handle(err)
+}
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+	// We can remove the isHelpRequest function when we have full Kong support for all commands
+	if isHelpRequest() {
+		return runCobraSystem()
+	}
+
+	cli := &CLI{}
+	parser, err := kong.New(cli, kong.Description("Buildkite CLI"), kong.NoDefaultHelp())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		return 1
+	}
+
+	ctx, err := parser.Parse(os.Args[1:])
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		return 1
+	}
+
+	if err := ctx.Run(cli); err != nil {
+		handleError(err)
+		return 1
+	}
+	return 0
+}
+
+// We can rip this out when we have full Kong support for all commands
+func isHelpRequest() bool {
+	if len(os.Args) < 2 {
+		return false
+	}
+
+	// Global help, e.g. bk --help
```

Comment:
We should also print help when no arguments are provided to match current behavior.